src/System.Threading.Tasks/tests/AggregateExceptionTests.cs

```diff
@@ -0,0 +1,157 @@
+            Assert.Equal(ex.GetBaseException(), ex);
+
+            innerExceptions = new Exception[1] { new AggregateException() };
+            ex = new AggregateException(innerExceptions);
+            Assert.Equal(ex.GetBaseException(), innerExceptions[0]);
+
+            innerExceptions = new Exception[2] { new AggregateException(), new AggregateException() };
+            ex = new AggregateException(innerExceptions);
+            Assert.Equal(ex.GetBaseException(), ex);
+        }
+
+        [Fact]
+        public static void RunAggregateException_Handle()
+        {
+            AggregateException ex = new AggregateException();
+            ex = new AggregateException(new[] { new ArgumentException(), new ArgumentException(), new ArgumentException() });
+            int handledCount = 0;
+            ex.Handle((e) =>
+            {
+                if (e is ArgumentException)
+                {
+                    handledCount++;
+                    return true;
+                }
+                return false;
+            });
+            Assert.Equal(handledCount, ex.InnerExceptions.Count);
+        }
+
+        [Fact]
+        public static void RunAggregateException_Handle_Negative()
+        {
+            AggregateException ex = new AggregateException();
+            Assert.Throws<ArgumentNullException>(() => ex.Handle(null));
+
+            ex = new AggregateException(new[] { new Exception(), new ArgumentException(), new ArgumentException() });
+            int handledCount = 0;
+            Assert.Throws<AggregateException>(
+               () => ex.Handle((e) =>
+               {
+                   if (e is ArgumentException)
+                   {
+                       handledCount++;
+                       return true;
+                   }
+                   return false;
+               }));
+        }
+
+        // Validates that flattening (incl recursive) works.
+        [Fact]
+        public static void RunAggregateException_Flatten()
+        {
+            Exception exceptionA = new Exception("A");
+            Exception exceptionB = new Exception("B");
+            Exception exceptionC = new Exception("C");
+
+            AggregateException aggExceptionBase = new AggregateException(exceptionA, exceptionB, exceptionC);
+
+            // Verify flattening one with another.
+            // > Flattening (no recursion)...
+
+            AggregateException flattened1 = aggExceptionBase.Flatten();
+            Exception[] expected1 = new Exception[] {
+                exceptionA, exceptionB, exceptionC
+            };
+
+            if (expected1.Length != flattened1.InnerExceptions.Count)
+            {
+                Assert.True(false, string.Format("RunAggregateException_Flatten: > error: expected count {0} differs from actual {1}",
+                    expected1.Length, flattened1.InnerExceptions.Count));
+            }
+
+            for (int i = 0; i < flattened1.InnerExceptions.Count; i++)
```

Comment:
Could this whole loop (and the previous assert) just be Assert(expected1, flattened1.InnerExceptions)?